bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/discovery/FreeboxOsDiscoveryService.java

```diff
@@ -107,49 +107,50 @@ protected void startScan() {
-                discoverPlugs(thingHandler.getManager(FreeplugManager.class), bridgeUID);
-                discoverRepeater(thingHandler.getManager(RepeaterManager.class), bridgeUID, lanHosts);
-                discoverPlayer(thingHandler.getManager(PlayerManager.class), bridgeUID, lanHosts);
-                discoverVM(thingHandler.getManager(VmManager.class), bridgeUID, lanHosts);
-                discoverHome(thingHandler.getManager(HomeManager.class), bridgeUID);
+                discoverServer(bridgeUID);
+                discoverPhone(bridgeUID);
+                discoverPlugs(bridgeUID);
+                discoverRepeater(bridgeUID, lanHosts);
+                discoverPlayer(bridgeUID, lanHosts);
+                discoverVM(bridgeUID, lanHosts);
+                discoverHome(bridgeUID);
                 if (thingHandler.getConfiguration().discoverNetDevice) {
-                    discoverHosts(thingHandler, bridgeUID, lanHosts);
+                    discoverHosts(bridgeUID, lanHosts);
                 }
             } catch (FreeboxException e) {
                 logger.warn("Error while requesting data for things discovery: {}", e.getMessage());
             }
         }
     }
 
-    private void discoverHome(HomeManager homeManager, ThingUID bridgeUID) throws FreeboxException {
+    private void discoverHome(ThingUID bridgeUID) throws FreeboxException {
         NodeConfigurationBuilder builder = NodeConfigurationBuilder.getInstance();
         try {
-            homeManager.getHomeNodes().forEach(
+            thingHandler.getManager(HomeManager.class).getHomeNodes().forEach(
                     node -> builder.configure(bridgeUID, node).ifPresent(result -> thingDiscovered(result.build())));
-        } catch (PermissionException e) {
-            logger.warn("Missing permission to discover Home {}", e.getPermission());
+        } catch (FreeboxException e) {
+            logger.warn("Error discovering Home: {}", e.getMessage());
         }
     }
 
-    private void discoverPlugs(FreeplugManager freeplugManager, ThingUID bridgeUID) {
+    private void discoverPlugs(ThingUID bridgeUID) {
         FreeplugConfigurationBuilder builder = FreeplugConfigurationBuilder.getInstance();
         try {
-            freeplugManager.getPlugs().forEach(plug -> thingDiscovered(builder.configure(bridgeUID, plug).build()));
+            thingHandler.getManager(FreeplugManager.class).getPlugs()
+                    .forEach(plug -> thingDiscovered(builder.configure(bridgeUID, plug).build()));
         } catch (FreeboxException e) {
-            logger.warn("Error discovering freeplugs {}", e.getMessage());
+            logger.warn("Error discovering freeplugs: {}", e.getMessage());
         }
     }
 
-    private void discoverPhone(PhoneManager phoneManager, ThingUID bridgeUID) throws FreeboxException {
+    private void discoverPhone(ThingUID bridgeUID) throws FreeboxException {
```

Comment:
The exception is caught within the method itself and not re-thrown, so it can be removed from the method signature.
```suggestion
    private void discoverPhone(ThingUID bridgeUID) {
```

Reply:
Fixed